apps/docs/docs/dev/12-jayvee-testing.md

```diff
@@ -0,0 +1,247 @@
+
+  const readJvTestAsset = readJvTestAssetHelper(
+    __dirname,
+    '../../test/assets/', // localized path to test assets folder
+  );
+
+  beforeAll(() => {
+    // [...] register extensions etc
+    const services = createJayveeServices(NodeFileSystem).Jayvee; // Or retrieve them if services already exist
+    // Parse function for Jayvee (without validation)
+    parse = parseHelper(services);
+  });
+
+  // [...]
+
+  it('My dummy test', () => {
+    const text = readJvTestAsset('<sub-folder>/<test-asset-name>.jv');
+
+    const document = await parse(text);
+    expectNoParserAndLexerErrors(document);
+    // Rest of test
+  });
+});
+```
+If you want to simply validate the test assets, simply replace `parseHelper` with `validationHelper` (and adjust the types).  
+For more information about this function, as well as all the others, please take a look at the file.
+
+[**extension/**](https://github.com/jvalue/jayvee/tree/main/libs/language-server/src/test/extension):  
+This folder contains an extension for testing. If there are certain blocks required for testing a certain feature, they can be defined here. One such example is the already defined `TestProperty` block which has a multitude of different properties, each with a different type. This block is used for testing properties and property-assignments.  
+The extension also already provides simple (meaning they don't have any properties) loader and extractor blocks for all IOTypes. These blocks are automatically generated at runtime with the following naming scheme: `Test${ioType}${io === 'input' ? 'Loader' : 'Extractor'}` (Example: `TestFileExtractor`).  
+This allows for easy (grammar) testing of non loader/extractor blocks:
+``` jv
+pipeline Pipeline {
+  block Test oftype CellWriter {
+    at: range A1:A3;
+    write: ['values', 'to', 'write'];
+  }
+
+  block TestExtractor oftype TestSheetExtractor {
+  }
+
+  block TestLoader oftype TestSheetLoader {
+  }
+
+  TestExtractor -> Test -> TestLoader;
+}
+```
+
+### Existing tests
+Currently there are already tests for the following parts:
+- Language-server validation checks (located [here](https://github.com/jvalue/jayvee/tree/main/libs/language-server/src/lib/validation))
+- Language-server constraint validation (located [here](https://github.com/jvalue/jayvee/tree/dev/libs/language-server/src/lib/constraint))
+- Custom block (property) validation of the three existing extensions (std extension located [here](https://github.com/jvalue/jayvee/blob/dev/libs/extensions/std/lang/src))
+- Grammar validation tests for all official full examples from the [/example](https://github.com/jvalue/jayvee/tree/main/example) folder (located [here](https://github.com/jvalue/jayvee/blob/dev/libs/extensions/std/lang/src/example-validation.spec.ts))
+- Grammar validation tests for all block examples of the std extension (located [here](https://github.com/jvalue/jayvee/blob/dev/libs/extensions/std/lang/src/meta-inf-example-validation.spec.ts))
```

Comment:
Not sure if we need such a list? Seems high maintenance for me.

Reply:
Yeah, but I think you mentioned that I should somehow/somewhere document what tests already exist

Reply:
Fair enough, let's keep them.